### PR TITLE
Vectorized Vector Dot Product node

### PR DESCRIPTION
**Pull request by harisreedhar**

### CODE REVIEW

File: animation_nodes/nodes/vector/c_utils.pyx [hunk 1]
```diff
@@ -105,3 +105,11 @@ def offset3DVectors(Vector3DList vectors, VirtualVector3DList offsets, FloatList
         vectors.data[i].x += offset.x * influence
         vectors.data[i].y += offset.y * influence
         vectors.data[i].z += offset.z * influence
+
+def calculateVectorDotProducts(VirtualVector3DList vectors1, VirtualVector3DList vectors2):
+    cdef Py_ssize_t i
+    cdef Py_ssize_t amount = VirtualVector3DList.getMaxRealLength(vectors1, vectors2)
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/vector/c_utils.pyx:111]: Compute the amount in the python code and pass it.

File: animation_nodes/nodes/vector/vector_dot_product.py [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
+            ("Dot Product", "dotProduct"), ("Dot Product", "dotProducts")))
 
     def getExecutionCode(self, required):
-        return "dotProduct = a.dot(b)"
+        if self.useListA or self.useListB:
+            yield "_vA = VirtualVector3DList.create(a, (0, 0, 0))"
+            yield "_vB = VirtualVector3DList.create(b, (0, 0, 0))"
+            yield "dotProducts = AN.nodes.vector.c_utils.calculateVectorDotProducts(_vA, _vB)"
+        else:
+            yield "distance = (a - b).length"
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/vector/vector_dot_product.py:26]: This line seems to be unused.

File: animation_nodes/nodes/vector/vector_dot_product.py [hunk 3]
```diff
@@ -1,14 +1,27 @@
...
+            ("B", "b"), ("B", "b")))
+
+        self.newOutput(VectorizedSocket("Float", ["useListA", "useListB"],
+            ("Dot Product", "dotProduct"), ("Dot Product", "dotProducts")))
 
     def getExecutionCode(self, required):
-        return "dotProduct = a.dot(b)"
+        if self.useListA or self.useListB:
+            yield "_vA = VirtualVector3DList.create(a, (0, 0, 0))"
+            yield "_vB = VirtualVector3DList.create(b, (0, 0, 0))"
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/vector/vector_dot_product.py:23]: Use the `createMultiple` method.

**Review by OmarEmaraDev — CHANGES_REQUESTED** [review]

File: animation_nodes/nodes/vector/c_utils.pyx (hunk 1 above)
Comment (harisreedhar) [reply animation_nodes/nodes/vector/c_utils.pyx]: Done

Comment (harisreedhar) [reply animation_nodes/nodes/vector/c_utils.pyx]: I have seen many times we compute amount outside. Is there any reason for that?

File: animation_nodes/nodes/vector/vector_dot_product.py (hunk 2 above)
Comment (harisreedhar) [reply animation_nodes/nodes/vector/vector_dot_product.py]: Done

File: animation_nodes/nodes/vector/vector_dot_product.py (hunk 3 above)
Comment (harisreedhar) [reply animation_nodes/nodes/vector/vector_dot_product.py]: Done

**Review by OmarEmaraDev — APPROVED** [review]
Thanks!

**Pull request merged**